tests/integration/test_cert_relation.py

```diff
@@ -85,16 +90,15 @@ async def build_and_deploy(
     relation_name = f"{INGRESS_APP_NAME}:nginx-route"
     await model.add_relation(f"{ANY_APP_NAME}:nginx-route", relation_name)
     await model.wait_for_idle()
-    provider_charm = await ops_test.build_charm(f"{PROVIDER_CHARM_DIR}/")
     await model.deploy(
-        provider_charm,
+        SELF_SIGNED_CERTIFICATES_CHARM_NAME,
         application_name=TLS_CERTIFICATES_PROVIDER_APP_NAME,
-        series="jammy",
+        channel="stable",
     )
```

Comment:
It seems that the `self-signed-certificates` on `latest/stable` is revision 155, which still uses version 3 of the TLS charm library. As a result, it doesn't recognize the new CSR in relation application data instead of unit data introduced in version 4 of the TLS charm library.

I assume this will also cause problems for anyone upgrading the `nginx-ingress-integrator` without upgrading TLS provider charms like the Lego provider?